File: Packages/RideSharingFeature/Sources/RideSharingFeature/RideDestinationOptionsSheets.swift
import SwiftUI

struct RideDestinationSheetView: View {
    @ObservedObject var viewModel: RideSharingViewModel

    var body: some View {
        LiiveBottomSheet {
            HStack(alignment: .center) {
                Text("Where to?")
                    .liiveStyle(.title2)
                    .foregroundColor(LiiveColor.text)
                Spacer()
                Text("Now ▾")
                    .font(LiiveFont.sheetMetaSemibold)
                    .foregroundColor(LiiveColor.accent)
            }
            .padding(.bottom, RideSheetLayout.headerBottomPadding)

            HStack(spacing: RideSheetLayout.rowGap) {
                Image(systemName: "magnifyingglass")
                    .font(.system(size: RideSheetLayout.searchIconSize, weight: .semibold))
                    .foregroundColor(LiiveColor.textSecondary)
                Text("Search a place or address")
                    .font(LiiveFont.callout)
                    .foregroundColor(LiiveColor.textTertiary)
                Spacer()
            }
            .frame(height: RideSheetLayout.searchHeight)
            .padding(.horizontal, RideSheetLayout.searchHorizontalPadding)
            .background(LiiveColor.fillTertiary)
            .clipShape(RoundedRectangle(cornerRadius: LiiveRadius.md, style: .continuous))
            .padding(.bottom, RideSheetLayout.sectionGap)

            VStack(spacing: RideSheetLayout.stackedSpacing) {
                ForEach(Array(RideFixtures.destinations.enumerated()), id: \.element.id) { index, place in
                    LiiveListRow(
                        title: place.title,
                        subtitle: place.subtitle,
                        divider: index < RideFixtures.destinations.count - 1,
                        chevron: true,
                        action: { viewModel.handle(.selectDestination(place)) }
                    ) {
                        LiiveIconCircle(
                            systemName: place.systemImage,
                            color: place.semanticColor.iconColor,
                            size: RideSheetLayout.savedPlaceIconSize
                        )
                    }
                }
            }
            .background(LiiveColor.surfaceRaised)
            .clipShape(RoundedRectangle(cornerRadius: LiiveRadius.lg, style: .continuous))
        }
        .accessibilityIdentifier(RideAccessibilityIdentifier.destinationSheet)
    }
}

struct RideOptionsSheetView: View {
    @ObservedObject var viewModel: RideSharingViewModel

    private var state: RideUIState { viewModel.state }
    private var selectedTier: RideTier { state.config.tier }

    var body: some View {
        LiiveBottomSheet {
            header
            tierPicker
            rideDetails
            LiiveButton(
                "Confirm Pickup · \(selectedTier.price.ridePrice)",
                variant: .primary,
                size: .lg,
                shape: .capsule,
                fullWidth: true,
                tabularNumbers: true
            ) {
                viewModel.handle(.confirmPickup)
            }
        }
        .accessibilityIdentifier(RideAccessibilityIdentifier.optionsSheet)
    }

    private var header: some View {
        HStack(spacing: RideSheetLayout.rowGap) {
            Button(action: { viewModel.handle(.backToDestination) }) {
                Image(systemName: "chevron.left")
                    .font(.system(size: RideSheetLayout.backIconSize, weight: .semibold))
                    .foregroundColor(LiiveColor.text)
                    .frame(width: RideSheetLayout.backButtonSize, height: RideSheetLayout.backButtonSize)
                    .background(LiiveColor.fillTertiary)
                    .clipShape(Circle())
            }
            .buttonStyle(.plain)

            VStack(alignment: .leading, spacing: RideSheetLayout.compactGap) {
                Text("Choose your ride")
                    .liiveStyle(.title3)
                    .foregroundColor(LiiveColor.text)
                Text("to \(state.destination?.title ?? "Union Square")")
                    .font(LiiveFont.footnote)
                    .foregroundColor(LiiveColor.textSecondary)
            }
            Spacer()
        }
        .padding(.bottom, RideSheetLayout.headerBottomPadding)
    }

    private var tierPicker: some View {
        VStack(spacing: RideSheetLayout.tierSpacing) {
            ForEach(RideTier.allCases) { tier in
                RideTierRow(tier: tier, isSelected: tier == selectedTier) {
                    viewModel.handle(.selectTier(tier))
                }
            }
        }
        .padding(.bottom, RideSheetLayout.sectionGap)
    }

    private var rideDetails: some View {
        VStack(spacing: RideSheetLayout.stackedSpacing) {
            LiiveListRow(title: "Passengers") {
                LiiveIconCircle(systemName: "person.2.fill", color: .neutral, size: RideSheetLayout.optionIconSize)
            } trailing: {
                LiiveStepper(value: state.config.passengers, range: 1...4) {
                    viewModel.handle(.setPassengers($0))
                }
            }
            LiiveListRow(title: "Bags") {
                LiiveIconCircle(systemName: "suitcase.fill", color: .neutral, size: RideSheetLayout.optionIconSize)
            } trailing: {
                LiiveStepper(value: state.config.bags, range: 0...4) {
                    viewModel.handle(.setBags($0))
                }
            }
            LiiveListRow(title: "Female-only pool", subtitle: "Match same-gender drivers & riders") {
                LiiveIconCircle(systemName: "shield.lefthalf.filled", color: .success, size: RideSheetLayout.optionIconSize)
            } trailing: {
                LiiveSwitch(isOn: Binding(
                    get: { state.config.femaleOnly },
                    set: { viewModel.handle(.setFemaleOnly($0)) }
                ))
            }
            LiiveListRow(title: "Child seat", divider: false) {
                LiiveIconCircle(systemName: "figure.child", color: .neutral, size: RideSheetLayout.optionIconSize)
            } trailing: {
                LiiveSwitch(isOn: Binding(
                    get: { state.config.childSeat },
                    set: { viewModel.handle(.setChildSeat($0)) }
                ))
            }
        }
        .background(LiiveColor.surfaceRaised)
        .clipShape(RoundedRectangle(cornerRadius: LiiveRadius.lg, style: .continuous))
        .padding(.bottom, RideSheetLayout.sectionGap)
    }
}

private struct RideTierRow: View {
    let tier: RideTier
    let isSelected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: RideSheetLayout.headerBottomPadding) {
                LiiveIconCircle(systemName: tier.systemImage, color: isSelected ? .accent : .neutral)
                VStack(alignment: .leading, spacing: RideSheetLayout.compactGap + RideSheetLayout.compactGap / 2) {
                    HStack(spacing: RideSheetLayout.inlineGap) {
                        Text(tier.name)
                            .font(LiiveFont.headline)
                            .foregroundColor(LiiveColor.text)
                        if tier.isMultiLeg {
                            LiiveBadge("2 legs", color: .warning)
                        }
                    }
                    Text(tier.detail)
                        .font(LiiveFont.footnote)
                        .foregroundColor(LiiveColor.textSecondary)
                }
                Spacer()
                VStack(alignment: .trailing, spacing: RideSheetLayout.compactGap) {
                    Text(tier.price.ridePrice)
                        .font(LiiveFont.headline.monospacedDigit())
                        .fontWeight(.bold)
                        .foregroundColor(LiiveColor.text)
                    Text(tier.eta)
                        .font(LiiveFont.caption1.monospacedDigit())
                        .foregroundColor(LiiveColor.textSecondary)
                }
            }
            .padding(RideSheetLayout.tierRowPadding)
            .background(LiiveColor.surfaceRaised)
            .overlay(
                RoundedRectangle(cornerRadius: LiiveRadius.lg, style: .continuous)
                    .strokeBorder(isSelected ? LiiveColor.accent : .clear, lineWidth: RideSheetLayout.selectedBorderWidth)
            )
            .clipShape(RoundedRectangle(cornerRadius: LiiveRadius.lg, style: .continuous))
        }
        .buttonStyle(.plain)
    }
}

extension Double {
    var ridePrice: String {
        String(format: "$%.2f", self)
    }
}

File: android/app/src/main/java/com/liive/ride/ui/RideDestinationOptionsSheets.kt
package com.liive.ride.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.liive.ride.*
import com.liive.ride.designsystem.*

@Composable
fun RideDestinationSheet(onEvent: (RideEvent) -> Unit) {
    val c = LiiveTheme.colors
    LiiveBottomSheet(modifier = Modifier.testTag(RideTestTags.DestinationSheet)) {
        Row(Modifier.fillMaxWidth().padding(bottom = 12.dp), verticalAlignment = Alignment.CenterVertically) {
            Text("Where to?", color = c.text, style = MaterialTheme.typography.headlineMedium)
            Spacer(Modifier.weight(1f))
            Text(
                "Now ▾",
                color = c.accent,
                style = MaterialTheme.typography.titleMedium.copy(fontSize = 14.sp),
                fontWeight = FontWeight.SemiBold
            )
        }

        Row(
            Modifier.fillMaxWidth().height(46.dp).clip(LiiveRadius.md).background(c.fillTertiary).padding(horizontal = 14.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            Icon(painterResource(RideIcons.Search), null, tint = c.textSecondary, modifier = Modifier.size(18.dp))
            Text("Search a place or address", color = c.textTertiary, style = MaterialTheme.typography.bodyMedium)
        }

        Spacer(Modifier.height(14.dp))
        Column(Modifier.fillMaxWidth().clip(LiiveRadius.lg).background(c.surfaceRaised)) {
            RideFixtures.destinations.forEachIndexed { index, place ->
                LiiveListRow(
                    title = place.title,
                    subtitle = place.subtitle,
                    divider = index < RideFixtures.destinations.lastIndex,
                    chevron = true,
                    leading = { LiiveIconCircle(place.icon, color = place.color, size = 36.dp) },
                    onClick = { onEvent(RideEvent.SelectDestination(place)) }
                )
            }
        }
    }
}

@Composable
fun RideOptionsSheet(state: RideUiState, onEvent: (RideEvent) -> Unit) {
    val c = LiiveTheme.colors
    LiiveBottomSheet(modifier = Modifier.testTag(RideTestTags.OptionsSheet)) {
        Row(Modifier.fillMaxWidth().padding(bottom = 12.dp), verticalAlignment = Alignment.CenterVertically) {
            Box(
                Modifier.size(32.dp).clip(LiiveRadius.full).background(c.fillTertiary)
                    .clickableNoRipple { onEvent(RideEvent.BackToDestination) },
                contentAlignment = Alignment.Center
            ) {
                Icon(painterResource(RideIcons.ChevronLeft), null, tint = c.text, modifier = Modifier.size(20.dp))
            }
            Spacer(Modifier.width(10.dp))
            Column(Modifier.weight(1f)) {
                Text("Choose your ride", color = c.text, style = MaterialTheme.typography.headlineSmall)
                Text("to ${state.destination?.title ?: "Union Square"}", color = c.textSecondary, style = MaterialTheme.typography.bodySmall)
            }
        }

        Column(verticalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.padding(bottom = 14.dp)) {
            RideTier.entries.forEach { tier ->
                RideTierRow(tier = tier, selected = tier == state.config.tier) { onEvent(RideEvent.SelectTier(tier)) }
            }
        }

        Column(Modifier.fillMaxWidth().clip(LiiveRadius.lg).background(c.surfaceRaised)) {
            LiiveListRow(
                title = "Passengers",
                leading = { LiiveIconCircle(RideFixtures.passengerIcon, IconCircleColor.Neutral, 32.dp) },
                trailing = { LiiveStepper(state.config.passengers, 1..4) { onEvent(RideEvent.SetPassengers(it)) } }
            )
            LiiveListRow(
                title = "Bags",
                leading = { LiiveIconCircle(RideFixtures.bagIcon, IconCircleColor.Neutral, 32.dp) },
                trailing = { LiiveStepper(state.config.bags, 0..4) { onEvent(RideEvent.SetBags(it)) } }
            )
            LiiveListRow(
                title = "Female-only pool",
                subtitle = "Match same-gender drivers & riders",
                leading = { LiiveIconCircle(RideFixtures.safetyIcon, IconCircleColor.Success, 32.dp) },
                trailing = { LiiveSwitch(state.config.femaleOnly) { onEvent(RideEvent.SetFemaleOnly(it)) } }
            )
            LiiveListRow(
                title = "Child seat",
                divider = false,
                leading = { LiiveIconCircle(RideFixtures.childSeatIcon, IconCircleColor.Neutral, 32.dp) },
                trailing = { LiiveSwitch(state.config.childSeat) { onEvent(RideEvent.SetChildSeat(it)) } }
            )
        }
        Spacer(Modifier.height(14.dp))
        LiiveButton(
            title = "Confirm Pickup · ${state.config.tier.price.ridePrice()}",
            onClick = { onEvent(RideEvent.ConfirmPickup) },
            fullWidth = true,
            size = LiiveButtonSize.Lg,
            capsule = true,
            tabularNumbers = true
        )
    }
}

@Composable
private fun RideTierRow(tier: RideTier, selected: Boolean, onClick: () -> Unit) {
    val c = LiiveTheme.colors
    Row(
        Modifier.fillMaxWidth()
            .clip(LiiveRadius.lg)
            .background(c.surfaceRaised)
            .border(1.5.dp, if (selected) c.accent else Color.Transparent, LiiveRadius.lg)
            .clickableNoRipple(onClick)
            .padding(12.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        LiiveIconCircle(tier.icon, color = if (selected) IconCircleColor.Accent else IconCircleColor.Neutral)
        Column(Modifier.weight(1f)) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(tier.displayName, color = c.text, style = MaterialTheme.typography.titleLarge)
                if (tier.multiLeg) LiiveBadge("2 legs", BadgeColor.Warning)
            }
            Text(tier.detail, color = c.textSecondary, style = MaterialTheme.typography.bodySmall)
        }
        Column(horizontalAlignment = Alignment.End) {
            Text(
                tier.price.ridePrice(),
                color = c.text,
                style = MaterialTheme.typography.titleLarge.tabularNumbers(),
                fontWeight = FontWeight.Bold
            )
            Text(tier.eta, color = c.textSecondary, style = MaterialTheme.typography.labelMedium.tabularNumbers())
        }
    }
}
